src/materialized/src/http/prof.rs

```diff
@@ -259,6 +262,42 @@ mod enabled {
                     .body(Body::from(s))
                     .unwrap())
             }
+            "dump_symbolicated_file" => {
+                let mut borrow = prof_ctl.lock().await;
+                let f = borrow.dump()?;
+                let r = BufReader::new(f);
+                let stacks = parse_jeheap(r)?;
+                let syms = symbolicate(&stacks);
+                let mut s = String::new();
+                // Emitting the format expected by Brendan Gregg's flamegraph tool.
+                //
+                // foo;bar;quux 30
+                // foo;bar;asdf 40
+                //
+                // etc.
+                for (stack, _anno) in stacks.iter() {
+                    for (i, addr) in stack.addrs.iter().enumerate() {
+                        let syms = match syms.get(addr) {
+                            Some(syms) => Cow::Borrowed(syms),
+                            None => Cow::Owned(vec!["???".to_string()]),
+                        };
+                        for (j, sym) in syms.iter().enumerate() {
+                            if j != 0 || i != 0 {
+                                s.push_str(";");
+                            }
+                            s.push_str(&*sym);
+                        }
+                    }
+                    writeln!(&mut s, " {}", stack.weight).unwrap();
+                }
+                Ok(Response::builder()
+                    .header(
+                        header::CONTENT_DISPOSITION,
+                        "attachement; filename=\"mz.fg\"",
```

Comment:
typo: `attachement` -> `attachment`. And, while you're changing this line, even though using raw strings is longer they might be easier to read? `r#"attachment; filename="mz.fg""#` , idk might be equivalent.

Reply:
whoops, I temporarily became French.

Reply:
btw, I don't really think the `r#` version is easier to read -- personal taste, I suppose.